api/mod/src/models/user.py

```diff
@@ -0,0 +1,35 @@
+"""
+Provides classes for Users
+"""
+
+from pydantic import BaseModel, Field
+
+
+class UserBase(BaseModel):
+    """Basic information about a user"""
+
+    username: str = Field(
+        ..., description="The name of the user", example="team-folivora"
+    )
+    email: str = Field(
+        ..., description="The email of the user", example="annoto@team-folivora.com"
+    )
+
+
+class UserCreate(UserBase):
+    """The fields required to create a new user"""
+
+    password: str = Field(
+        ..., description="The password of the user", example="test1234"
+    )
+
+
+class User(UserBase):
+    """An already existing user"""
+
+    id: int = Field(..., description="The id of the user", example=1)
+
+    class Config:
+        """Pydanic configuration"""
+
+        orm_mode = True
```

Comment:
I think the idea was, that we never want to send passwords or even the hashed version to the frontend but want to keep it in the backend or even better only in the database. And this behavior is ensured through design here.
And this is how the fastapi docs recommended it, so this might add some importance 😅

Reply:
Yes, totally agree with you, we should not send passwords to the frontend! (this was never my intention 😅)
For me, it's more of a naming problem, because "User" sounds like he would have averything related to users to me
But I'm also fine leaving it as is